methods/guest/src/bin/email_verify.rs

```diff
@@ -0,0 +1,49 @@
+#![no_main]
+
+use cfdkim::{ verify_email_with_key, DkimPublicKey };
+use mailparse::parse_mail;
+use risc0_zkvm::guest::env;
+use sha2::{ Digest, Sha256 };
+use slog::{ o, Discard, Logger };
+use zkemail_core::{ EmailVerifierOutput, Email };
+
+fn hash_bytes(data: &[u8]) -> Vec<u8> {
+    let mut hasher = Sha256::new();
+    hasher.update(data);
+    hasher.finalize().to_vec()
+}
+
+fn verify_dkim(input: &Email, logger: &Logger) -> bool {
```

Comment:
verify_dkim can be put in a separate file and be used by both `email_verify.rs` and `email_with_regex_verify.rs`